checks/raw/pinned_dependencies_test.go

```diff
@@ -415,6 +416,75 @@ func TestDockerfilePinning(t *testing.T) {
 	}
 }
 
+func TestMixedPinning(t *testing.T) {
+	t.Parallel()
+
+	workflowDependency := checker.Dependency{
+		Name:     stringAsPointer("actions/checkout"),
+		PinnedAt: stringAsPointer("daadedc81d5f9d3c06d2c92f49202a3cc2b919ba"),
+		Location: &checker.File{
+			Path:      ".github/workflows/workflow-not-pinned.yaml",
+			Snippet:   "actions/checkout@daadedc81d5f9d3c06d2c92f49202a3cc2b919ba",
+			Offset:    31,
+			EndOffset: 31,
+			FileSize:  0,
+			Type:      1,
+		},
+		Pinned:      boolAsPointer(false),
```

Comment:
This looks pinned due to the `Snippet` and the `PinnedAt`, but the `Path` and `Pinned` fields seem to conflict with that.